callisto/delivery/matching.py

```diff
@@ -1,74 +1,108 @@
-    checked for matches.
-
-    Args:
-      identifiers (list of str): the new identifiers to check for matches
-    """
-    for identifier in identifiers:
-        match_list = [potential for potential in MatchReport.objects.all() if potential.get_match(identifier)]
-        if len(match_list) > 1:
-            seen_match_owners = [match.report.owner for match in match_list if match.seen]
-            new_match_owners = [match.report.owner for match in match_list if not match.seen]
-            # filter out multiple reports made by the same person
-            if len(set(seen_match_owners + new_match_owners)) > 1:
-                # only send notifications if new matches are submitted by owners we don't know about
-                if not set(new_match_owners).issubset(set(seen_match_owners)):
-                    process_new_matches(match_list, identifier)
-                for match_report in match_list:
-                    match_report.report.match_found = True
-                    match_report.report.save()
-        for match in match_list:
-            match.seen = True
-            # delete identifier, which should only be filled for newly added match reports in delayed matching case
-            match.identifier = None
-            match.save()
-
-
-def process_new_matches(matches, identifier):
-    """Sends a report to the receiving authority and notifies the reporting users. Each user should only be notified
-    one time when a match is found.
-
-    Args:
-      matches (list of MatchReports): the MatchReports that correspond to this identifier
-      identifier (str): identifier associated with the MatchReports
-    """
-    logger.info("new match found")
-    owners_notified = []
-    for match_report in matches:
-        EvalRow.store_eval_row(action=EvalRow.MATCH_FOUND, report=match_report.report)
-        owner = match_report.report.owner
-        # only send notification emails to new matches
-        if owner not in owners_notified and not match_report.report.match_found \
-                and not match_report.report.submitted_to_school:
-            DeliveryApi().send_match_notification(owner, match_report)
-            owners_notified.append(owner)
-    # send report to school
-    DeliveryApi().send_matching_report_to_school(matches, identifier)
+class MatchingApi(Api):
+    '''
+        Used to route calls to from inside of callisto/delivery/* to matching api(s) inside of other apps
+
+        this object is called with
+            from callisto.delivery.matching import MatchingApi
+            MatchingApi().run_matching()
+
+        which maps to CALLISTO_MATCHING_API
+            CALLISTO_MATCHING_API.run_matching
+
+        use CALLISTO_MATCHING_API to override the class that MatchingApi maps to
+    '''
+
+    api_env_variable = 'CALLISTO_MATCHING_API'
+    default_classpath = 'callisto.delivery.matching.CallistoMatching'
+
+
+class CallistoMatching(object):
```

Comment:
changes in this class: all methods now take `MatchReports` instead of identifier strings, and addition of `get_all_eligible_match_reports`